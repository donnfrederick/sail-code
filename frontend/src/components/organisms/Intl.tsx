import React, {
  PropsWithChildren,
  Dispatch,
  SetStateAction,
  useContext,
  useState
} from 'react'
import { IntlProvider } from 'react-intl'
import { messages, browserLocale, Locale } from 'utils/intl'

export type LocaleState = {
  locale: Locale
}

const initialState: LocaleState = {
  locale: browserLocale
}

const IntlStateContext = React.createContext<LocaleState>(initialState)

const SetLocaleStateContext = React.createContext<
  Dispatch<SetStateAction<LocaleState>>
>(() => {
  // empty block
})

export function useLocaleState() {
  return useContext(IntlStateContext)
}
export function useSetLocaleState() {
  return useContext(SetLocaleStateContext)
}

export function LocaleStateProvider(props: {
  initialState?: LocaleState
  children: React.ReactNode
}) {
  const [state, setState] = useState<LocaleState>(
    props.initialState || initialState
  )
  return (
    <IntlStateContext.Provider value={state}>
      <SetLocaleStateContext.Provider value={setState}>
        {props.children}
      </SetLocaleStateContext.Provider>
    </IntlStateContext.Provider>
  )
}

export const Intl = (props: PropsWithChildren<{}>) => {
  const localeState = useLocaleState()

  return (
    <LocaleStateProvider>
      <IntlProvider
        messages={messages[localeState.locale]}
        locale={localeState.locale}
        defaultLocale="en"
      >
        {props.children}
      </IntlProvider>
    </LocaleStateProvider>
  )
}
