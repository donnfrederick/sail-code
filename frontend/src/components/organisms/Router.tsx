import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
