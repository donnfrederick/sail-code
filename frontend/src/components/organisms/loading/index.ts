import Loading from 'components/organisms/loading/Loading'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  isFetchingConversation: boolean
  isFetchingLocations: boolean
  isFetchingNotification: boolean
  isFetchingOrganizations: boolean
  isFetchingStudentsMe: boolean
  isFetchingTeachersMe: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    isFetchingConversation: state.rootReducer.conversation.fetching > 0,
    isFetchingLocations: state.rootReducer.locations.isFetching,
    isFetchingNotification: state.rootReducer.notification.isFetching,
    isFetchingOrganizations: state.rootReducer.organizations.isFetching,
    isFetchingStudentsMe: state.rootReducer.students.isFetching,
    isFetchingTeachersMe: state.rootReducer.teachers.isFetching
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {}
)

export default enhancer(Loading)
