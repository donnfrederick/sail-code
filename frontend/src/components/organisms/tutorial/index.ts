import Tutorial from 'components/organisms/tutorial/Tutorial'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as TutorialActions from 'reducers/tutorial'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  shouldShow: boolean
  showConfirmation: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    shouldShow: state.rootReducer.tutorial.shouldShow,
    showConfirmation: state.rootReducer.teachers.showConfirmation
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    hide: TutorialActions.hide,
    show: TutorialActions.show
  }
)

export default enhancer(Tutorial)
