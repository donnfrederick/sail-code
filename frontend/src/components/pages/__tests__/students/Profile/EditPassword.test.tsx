import Router from 'components/organisms/Router'
import StudentsProfileEditPassword from 'components/pages/students/Profile/EditPassword'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Intl } from 'components/organisms/Intl'

test('render <StudentsProfileEditPassword />', () => {
  const tree = renderer
    .create(
      <Intl>
        <Router>
          <StudentsProfileEditPassword />
        </Router>
      </Intl>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
