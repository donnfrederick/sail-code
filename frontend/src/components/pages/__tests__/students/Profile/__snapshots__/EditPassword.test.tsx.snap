// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <StudentsProfileEditPassword /> 1`] = `
<div
  className="sc-kEYyzF lhxmFk"
>
  <div
    className="sc-gqjmRU cTuUiK"
  >
    <div
      className="sc-gZMcBi UIhMm"
    >
      <div
        className="sc-iwsKbI dRoDuh"
      >
        Current Password
      </div>
      <input
        className="sc-dnqmqq cFKGVo"
        data-error={false}
        onInput={[Function]}
        placeholder="Current Password"
        type="password"
      />
      <div
        className="sc-htoDjs knnjmL"
      />
    </div>
    <div
      className="sc-gZMcBi UIhMm"
    >
      <div
        className="sc-iwsKbI dRoDuh"
      >
        New Password
      </div>
      <input
        className="sc-dnqmqq cFKGVo"
        data-error={false}
        onInput={[Function]}
        placeholder="New Password"
        type="password"
      />
      <div
        className="sc-htoDjs knnjmL"
      />
    </div>
    <div
      className="sc-VigVT leyFzv"
    >
      <button
        className="sc-bdVaJa evUcPv"
        data-active={false}
        fontSize={32}
        height={88}
        onClick={[Function]}
        type="blue"
        width={488}
      >
        Save
      </button>
    </div>
  </div>
  <header
    className="sc-kGXeez coovXG"
  >
    <div
      className="sc-kpOJdX dBkpMv"
    >
      Change Password
    </div>
    <div
      className="sc-dxgOiQ jnaKWi"
      onClick={[Function]}
    />
  </header>
  <div
    className="sc-gzVnrw bbOeQD"
  >
    <a
      className="sc-EHOje gEifQf"
      href="/students/mypage"
      icon="/assets/img/students/footer/home@3x.png"
      onClick={[Function]}
    />
    <a
      className="sc-EHOje bIkBAA"
      href="/students/notifications"
      icon="/assets/img/students/footer/notification@3x.png"
      onClick={[Function]}
    />
    <a
      className="sc-EHOje exXkhS"
      href="/students/mypage"
      icon="/assets/img/students/footer/cart.png"
      onClick={[Function]}
    />
    <a
      className="sc-bZQynM iAxnUN"
      href="/students/reservations/new"
      icon="/assets/img/students/footer/calendar@3x.png"
      onClick={[Function]}
    />
  </div>
</div>
`;
